pkg/controllers/management/auth/globalroles/enqueue.go

```diff
@@ -126,3 +129,61 @@ func (g *globalRBACEnqueuer) crtbEnqueueGRB(_, _ string, obj runtime.Object) ([]
 		{Name: grbOwner},
 	}, nil
 }
+
+// roleEnqueueGR enqueues a GlobalRole owns a given Role when that Role is changed. Uses the label
+// which is protected by the webhook rather than the ownerReference
+func (g *globalRBACEnqueuer) roleEnqueueGR(_, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
+	if obj == nil {
+		return nil, nil
+	}
+	role, ok := obj.(*v1.Role)
+	if !ok {
+		logrus.Errorf("unable to convert object: %[1]v, type: %[1]T to a Role", obj)
+		return nil, nil
+	}
+	grOwner, ok := role.Labels[grOwnerLabel]
+	if !ok {
+		// this Role isn't owned by a GR, no need to enqueue a GR
+		return nil, nil
+	}
+	_, err := g.grCache.Get(grOwner)
```

Comment:
Is there any way we can add tests to verify that this works correctly? Perhaps not restricted to the enqueue function but the core GR reconciliation. eg: Including tests with a very long named `Role`.

Reply:
Since these and the other handler which uses this item are sort of separate from one another, this would be something that would be better caught as part of an integration test (i.e. change the role make sure the owner was created, do it again with a longer name).

There's already a lot here, so I'm not eager to call that "in scope" for this PR, but that's my thought on where that would go.

Reply:
Ok.

It feels like our current unit tests don't cover enough (this OP comment kinda points to that) and integration tests are not favored/encouraged for many valid reasons. It would be good to find a solution in between for this (obviously out of scope of this PR).

As for tests that covers this, I'd call that "in scope" for this PR if we had an easy way to add them.. But it looks like we don't, so I won't push more for it. Sorry for derailing the first comment, which still needs fixing :+1: